src/seismometer/data/pandas_helpers.py

```diff
@@ -196,17 +223,34 @@ def infer_label(
     if label_col not in dataframe.columns or time_col not in dataframe.columns:
         return dataframe
 
-    try:  # Assume numeric labels: edge case Float and Int incompatibilities
-        dataframe[label_col] = dataframe[label_col].astype(float)
-    except BaseException:  # Leave as nonnumeric
-        pass
+    # use pandas for compatibility of imutations -- handle Nones
+    impute_val = pd.Series([impute_val_no_time or 0, impute_val_with_time or 1], dtype=(column_dtype or float)).values
```

Comment:
if we are going to use a Series for this, can we give the index nice names so that we can reference them in lines 234/238?

Reply:
ooh nice! using names that also eliminate comments